### PR TITLE
Resoves #977. Document how to compile a local asciidoctor ruby code base

### PR DESCRIPTION
**Pull request by djencks**

### CODE REVIEW

File: CONTRIBUTING-CODE.adoc [hunk 1]
```diff
@@ -77,6 +80,29 @@ For instance to build against Asciidoctor (Ruby) v2.0.10 use:
...
+The build instructions above supply this by unpacking a tar.gz of the repository constructed by GitHub.
+Alternatively, you can supply this copy yourself.
+For the following posix-centric suggestions, assume `ASCIIDOCTOR_RB` is the local location of your Asciidoctor Ruby project.
+
+* Remove the current (contents of the) Ruby source directory: `rm -rf packages/core/build/asciidoctor`
+* Choose a method:
+** Using a symlink: `ln -s $ASCIIDOCTOR_RB packages/core/build/asciidoctor`
+** Copying: `cp -r $ASCIIDOCTOR_RB packages/core/build/`
+** Unpacking a gem:
+----
```
Comment (Mogztter) [inline CONTRIBUTING-CODE.adoc:95]: We might use a list continuation to indent the listing block:

```suggestion
+
----
```

File: CONTRIBUTING-CODE.adoc [hunk 2]
```diff
@@ -77,6 +80,29 @@ For instance to build against Asciidoctor (Ruby) v2.0.10 use:
...
+mkdir packages/core/build/asciidoctor
+tar -xf $ASCIIDOCTOR_RB/pkg/asciidoctor-*.gem -C packages/core/build --include data.tar.gz
+tar -xzf packages/core/build/data.tar.gz -C packages/core/build/asciidoctor
+----
+* For all of these, `touch packages/core/build/asciidoctor.tar.gz` will prevent `npm build --prefix packages/core` from overwriting your copy with a downloaded one.
+* To actually use your local source you'll need to force recompile by:
+----
```
Comment (Mogztter) [inline CONTRIBUTING-CODE.adoc:102]: We might use a list continuation to indent the listing block:

ditto
```suggestion
+
----
```

File: CONTRIBUTING-CODE.adoc (hunk 1 above)
Comment (djencks) [reply CONTRIBUTING-CODE.adoc:108]: I liked that so much I changed most of the other usages!

**Comment by Mogztter** [comment]
Excellent thanks again! 👌

**Pull request merged**